File: docs/snippets/all/archetypes/graph_undirected.py
"""Log a simple undirected graph."""

import dalaran as dl

dl.init("dalaran_example_graph_undirected", spawn=True)

dl.log(
    "simple",
    dl.GraphNodes(
        node_ids=["a", "b", "c"],
        positions=[(0.0, 100.0), (-100.0, 0.0), (100.0, 0.0)],
        labels=["A", "B", "C"],
    ),
    dl.GraphEdges(
        edges=[("a", "b"), ("b", "c"), ("c", "a")],
        # Optional: graphs are undirected by default.
        graph_type="undirected",
    ),
)

File: docs/snippets/all/archetypes/graph_undirected.rs
//! Log a simple undirected graph.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec =
        dalaran::RecordingStreamBuilder::new("dalaran_example_graph_undirected")
            .spawn()?;

    rec.log(
        "simple",
        &[
            &dalaran::GraphNodes::new(["a", "b", "c"])
                .with_positions([(0.0, 100.0), (-100.0, 0.0), (100.0, 0.0)])
                .with_labels(["A", "B", "C"])
                as &dyn dalaran::AsComponents,
            &dalaran::GraphEdges::new([("a", "b"), ("b", "c"), ("c", "a")])
                // Optional: graphs are undirected by default.
                .with_undirected_edges(),
        ],
    )?;

    Ok(())
}
